components/PokemonList.tsx

```diff
@@ -56,13 +63,37 @@ function PokemonList({ id }: PokemonProps) {
     fairy: "bg-pink-300",
   };
 
-  const filteredSearch = allPokemon.filter((item) =>
-    item.name.toLowerCase().includes(input.toLowerCase())
+  const {
+    data: infiniteData,
+    isLoading: infiniteIsLoading,
+    fetchNextPage,
+  } = usePokemonInfinite();
+
+  const pokemon: pokemon[] | undefined = infiniteData?.pages.flatMap(
+    ({ content }) => content
   );
 
+  console.log(infiniteData);
+  const filteredSearch = allPokemon.filter((item) => {
+    const lowercaseName = item.name.toLowerCase();
+    const lowercaseInput = input.toLowerCase();
+    if (searchFilter === "Type") {
+      return item.types.some((type) =>
+        type.name.toLowerCase().includes(lowercaseInput)
+      );
+    } else if (searchFilter === "Ability") {
+      return item.abilities.some((ability) =>
+        ability.name.toLowerCase().includes(lowercaseInput)
+      );
+    } else if (searchFilter === "Name") {
+      return lowercaseName.includes(lowercaseInput);
+    } else {
+      return lowercaseName.includes(lowercaseInput);
+    }
```

Comment:
Else if blocks can benefit from using what's called the ["implicit return pattern"](https://medium.com/@rachelchervin/if-else-vs-if-return-using-implicit-returns-in-javascript-and-react-bbac63656e1b). When you return, you are ending the execution of that function scope. In other words, we don't need the else keyword. If a condition is met, simply return.